### ISSUE TITLE
Feature "Redact Email Addresses" doesn't work?

### ISSUE THREAD

**Issue by IGI2** [BoltAI/BoltAI#80]
Hi! 

<img width="488" alt="Conversation with emails" src="https://github.com/BoltAI/BoltAI/assets/39410473/ea22ff1d-b9a1-4d80-9e9d-96bec05cb583">

I don't think that the feature "Redact Email Addresses" (and similar) works properly, or I don't know how it should work. 

I thought, it should redact the email addresses before sending over the API, or when? From what I see, it keeps the email address in all cases, and at all points of working with the app.

**Comment by longseespace** [comment]
Hi @IGI2. Sorry for the confusion. The Data Redaction feature only applies to AI Command or AI Inline, not in the main chat.

This was a legacy design decision where there might be sensitive data in another app / document that you're not aware of.

When you actively share the email in the chat like in this case, I'm not sure if the redaction feature is useful.

Though I'm happy to change the behavior. What do you think would make more sense?

**Comment by IGI2** [comment]
I agree that adding another toggle could be a viable solution, one for AI Command/AI Inline and another for the main chat. This would provide users with more control over their privacy settings depending on the context.

Regarding the redaction of email addresses, I believe it would be more beneficial to partially redact them to preserve the context of the conversation. For instance, transforming "john@apple.com" to "john@example.com" maintains the user's name while obscuring the specific domain. Using "[example.com](http://example.com/)" or "[not-a-real-domain.com](http://not-a-real-domain.com/)" should be sufficient for understanding within the LLM, as these are widely recognized as placeholders and do not reveal actual company information. This approach balances privacy with maintaining enough detail for meaningful communication.

Ideally, I would go for obfuscating somehow the first name, but that might be too complex, and it should be consistent if we ever work with a different fragment where the same person appears. But maybe instead of complex obfuscating, we could leave first 5-7-10 characters? For example, john.giannandrea@apple.com -> john.gian@example.com
